samtranslator/model/sam_resources.py

```diff
@@ -802,6 +803,114 @@ def _validate_deployment_preference_and_add_update_policy(
             )
 
 
+class SamCanary(SamResourceMacro):
+    """SAM canary macro."""
+
+    resource_type = "AWS::Serverless::Canary"
+    property_types = {
+        "FunctionName": PropertyType(True, is_str()),
+        "Handler": PropertyType(True, is_str()),
+        "Runtime": PropertyType(True, is_str()),
+        "CodeUri": PropertyType(False, one_of(is_str(), is_type(dict))),
+        "InlineCode": PropertyType(False, one_of(is_str(), is_type(dict))),
+        "MemorySize": PropertyType(False, is_type(int)),
+        "Tags": PropertyType(False, is_type(dict)),
+        "Tracing": PropertyType(False, is_type(bool)),
+        "Timeout": PropertyType(False, is_type(int)),
+        "Role": PropertyType(True, is_str()),
```

Comment:
Do we want to make this mandatory, given we are going to resolve this if not provided?

Reply:
Made "Role" and "ArtifactS3Location" not required as they will be resolved if they are not provided